.github/workflows/ci.yml

```
@@ -0,0 +1,33 @@
+name: Build
+
+on:
+  push:
+    # dependabot will create both branches and pull request
+    # only run on pull requests to avoid running twice and creating a release draft
+    branches-ignore:
+      - 'dependabot/**'
+  pull_request:
```

Let's restrict this to PRs to the `main` branch and direct commits on the `main` branch.
```suggestion
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

---

The setting for "push" would prevent the action to run on forks where people create their feature branches. Therefore I'd like to keep the setting for "push" as is.

I'll add the setting for pull_requests though.